### DISCUSSION TITLE
How do I deal with the chicken and egg situation regarding prometheus integration?

### DISCUSSION THREAD

**Discussion by UntouchedWagons** [rook/rook#13786] [Q&A] [answered]
When creating a Ceph cluster I can enable prometheus integration via the monitoring subsection, but prometheus depends on storage to run which isn;t available until a Ceph cluster is created. I imagine deploying the cluster would fail because the templates would reference objects that don't exist because Prometheus isn't installed. How would I reconcile this?

**Comment by travisn** [comment, marked as answer]
Configuring prometheus just sets up the endpoints for Prometheus to scrape. So you should be able to bring Rook up first, then after Prometheus is enabled, it would start scraping. Beware that if there are any downtime issues in Rook, you won't be able to use the prometheus metrics to troubleshoot the issue, so it's generally recommended to have independent storage for Prometheus.
